vocabulary-care-pathways-questions/api/src/main/java/org/phenotips/vocabulary/internal/solr/CarePathwaysQuestionOntology.java

```diff
@@ -0,0 +1,350 @@
+import org.apache.commons.csv.CSVParser;
+import org.apache.commons.csv.CSVRecord;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.solr.client.solrj.SolrQuery;
+import org.apache.solr.client.solrj.SolrQuery.ORDER;
+import org.apache.solr.client.solrj.util.ClientUtils;
+import org.apache.solr.common.SolrInputDocument;
+import org.apache.solr.common.params.DisMaxParams;
+import org.apache.solr.common.params.SpellingParams;
+
+/**
+ * Provides access to the Care Pathways questions vocabulary. The vocabulary prefix is {@code CPQ}.
+ *
+ * @version $Id$
+ * @since 1.0
+ */
+@Component
+@Named("care-pathways-questions")
+@Singleton
+public class CarePathwaysQuestionOntology extends AbstractCSVSolrVocabulary
+{
+    /** The category for care pathways question-related terms. */
+    private static final String QUESTION_CATEGORY = "survey-question";
+
+    /** The standard prefix for the vocabulary terms. */
+    private static final String STANDARD_PREFIX = "CPQ";
+
+    /** A colon. */
+    private static final String COLON = ":";
+
+    /** The label for the field storing ID for term. */
+    private static final String ID = "id";
+
+    /** The label for the field storing the question string. */
+    private static final String NAME = "name";
+
+    /** The label for the field storing the question group. */
+    private static final String TERM_GROUP = "term_group";
+
+    /** For determining if a query is a an id. */
+    private static final Pattern ID_PATTERN = Pattern.compile("^CPQ:[0-9]+$", Pattern.CASE_INSENSITIVE);
+
+    @Override
+    protected int getSolrDocsPerBatch()
+    {
+        return 500000;
+    }
+
+    @Override
+    protected String getCoreName()
+    {
+        return getIdentifier();
+    }
+
+    @Override
+    public String getIdentifier()
+    {
+        return "care-pathways-questions";
+    }
+
+    @Override
+    public String getName()
+    {
+        return "Care pathways survey questions";
+    }
+
+    @Override
+    public Set<String> getAliases()
+    {
+        Set<String> aliases = new HashSet<>();
+        aliases.add(getIdentifier());
+        aliases.add(STANDARD_PREFIX);
+        return aliases;
+    }
+
+    @Override
+    public Collection<String> getSupportedCategories()
+    {
+        return Collections.singletonList(QUESTION_CATEGORY);
+    }
+
+    @Override
+    public String getDefaultSourceLocation()
+    {
+        return StringUtils.EMPTY;
+    }
+
+    @Override
+    public String getWebsite()
+    {
+        return StringUtils.EMPTY;
+    }
+
+    @Override
+    public String getCitation()
+    {
+        return StringUtils.EMPTY;
+    }
+
+    @Nullable
+    @Override
+    public VocabularyTerm getTerm(@Nullable final String id)
+    {
+        final VocabularyTerm result = super.getTerm(id);
+        if (result == null) {
+            final String optionalPrefix = STANDARD_PREFIX + COLON;
+            return StringUtils.startsWith(id, optionalPrefix)
```

Comment:
Can you tell me why this is needed?

Reply:
If someone accidentally ends up inputting something like  CPQ:CPQ:10. Probably not necessary, but i remember running into something like that somewhere some time ago. 

Should i remove?

Reply:
This code pattern exists for OMIM, but there it is needed because we actually indexed the terms without the `OMIM:` prefix. I'd say remove this, just let the inherited method implementation work as is.

Reply:
Fixed.